### PR TITLE
Fix repeated word typos

### PR DESCRIPTION
**Pull request by wikku**
Inspired by #57295 (I skipped 'be be' because of it) and my [PR in another repo
](https://github.com/e-maxx-eng/e-maxx-eng/pull/389)
Not a stupid `sed`, I actually tried to fix case by case.

### CODE REVIEW

File: src/librustc/ty/fold.rs [hunk 1]
```diff
@@ -110,7 +110,7 @@ pub trait TypeFoldable<'tcx>: fmt::Debug + Clone {
         self.has_type_flags(TypeFlags::HAS_FREE_REGIONS)
     }
 
-    /// True if there any any un-erased free regions.
+    /// True if there any un-erased free regions.
```
Comment (u32i64) [inline src/librustc/ty/fold.rs]: Shouldn't this be
```suggestion
    /// True if there are any un-erased free regions.
```

**Comment by u32i64** [comment]
Oh ok, `has has` is here just like `that that` in some other places. Hard to parse what it means at first.

I don't think it needs a comma though.

**Comment by Centril** [comment]
@bors r-

@wiktorkuchta If you want, add follow up commits to fix more typos. You can also file follow up PRs... :)
